app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/Creditmemos.php

```diff
@@ -14,6 +17,27 @@
 class Creditmemos extends \Magento\Framework\View\Element\Text\ListText implements
     \Magento\Backend\Block\Widget\Tab\TabInterface
 {
+     /**
+     * @var AuthorizationInterface
+     */
+    private $authorization;
+
+    /**
+     * Creditmemos constructor.
+     *
+     * @param Context $context
+     * @param AuthorizationInterface $authorization
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        AuthorizationInterface $authorization,
```

Comment:
This class is marked with `@api` class, so we can't introduce any backward incompatible changes.

Could you make this argument the last one, make it optional with fallback to retrieving this object from the object manager?

Reply:
@ihor-sviziev, I see your point.  
Do you have a link to some guide or documentation about why it should be done this way?  
I was not expected dependency injection parameters to be something that cannot be changed.  
And as for me, taking an object from the object manager braking other code style / architeture principles.

Also, I think it's a bit weird to have these classes marked as `@api`, maybe you are aware of a process or a case of removing `@api`from classes where it shouldn't be?

Reply:
You can find “adding constructor argument” section in https://devdocs.magento.com/contributor-guide/backward-compatible-development/
I don’t think some class could be in-marked as api, also it will be too long approval process that might take from few months to more than half a year

Reply:
Thanks, @ihor-sviziev   
I'll make backward-compatible

Reply:
Done